### ISSUE TITLE
keras to onnx conversion notebook is failing on Windows

### ISSUE THREAD

**Issue by cwmeijer** [dianna-ai/dianna#264]

**Comment by cwmeijer** [comment]
There is a problem running the notebook on Windows on the server, but it is running just fine on my machine (also with a fresh environment). 

### What I changed to debug
Because running a notebook using pytest doesn't result in output, I altered the notebook in PR #265. The problem is with running the conversion command line tool:
```python
onnx_savedmodel = 'resnet_savedmodel.onnx'
!python -m tf2onnx.convert --saved-model {savedmodel_dir} --output {onnx_savedmodel} --signature_def serving_default --tag serve
```
In order to have the output of that appear while running the test, I captured the output and raised an exception with it:
```python
onnx_savedmodel = 'resnet_savedmodel.onnx'
x = !python -m tf2onnx.convert --saved-model {savedmodel_dir} --output {onnx_savedmodel} --signature_def serving_default --tag serve

s = '\n'.join(x)
raise Exception(s)
```

### Results
Note that both print start with an Error, which is the one I deliberately raised in order to expose the output from the notebook to the log.
On my machine (Windows 10, Powershell, no gpu), conversion succeeds and the onnx file is written. I get the following output:
```python
Exception                                 Traceback (most recent call last)
Input In [6], in <module>
      1 s = '\n'.join(x)
----> 2 raise Exception(s)

Exception: 2022-03-10 10:36:13.989666: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2022-03-10 10:36:13.989781: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
C:\Users\ChristiaanMeijer\anaconda3\envs\dianna-x\lib\runpy.py:127: RuntimeWarning: 'tf2onnx.convert' found in sys.modules after import of package 'tf2onnx', but prior to execution of 'tf2onnx.convert'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
2022-03-10 10:36:17.129849: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2022-03-10 10:36:17.129888: W tensorflow/stream_executor/cuda/cuda_driver.cc:326] failed call to cuInit: UNKNOWN ERROR (303)
2022-03-10 10:36:17.134081: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: ESLT0114
2022-03-10 10:36:17.134216: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: ESLT0114
2022-03-10 10:36:17.134638: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-10 10:36:23,615 - INFO - Signatures found in model: [serving_default].
2022-03-10 10:36:23,616 - INFO - Output names: ['predictions']
2022-03-10 10:36:23.638858: I tensorflow/core/grappler/devices.cc:69] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
2022-03-10 10:36:23.639596: I tensorflow/core/grappler/clusters/single_machine.cc:357] Starting new session
2022-03-10 10:36:23.730507: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1144] Optimization results for grappler item: graph_to_optimize
  function_optimizer: Graph size after: 1200 nodes (877), 1855 edges (1532), time = 45.902ms.
  function_optimizer: function_optimizer did nothing. time = 0.818ms.

WARNING:tensorflow:From C:\Users\ChristiaanMeijer\anaconda3\envs\dianna-x\lib\site-packages\tf2onnx\tf_loader.py:706: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
2022-03-10 10:36:26,415 - WARNING - From C:\Users\ChristiaanMeijer\anaconda3\envs\dianna-x\lib\site-packages\tf2onnx\tf_loader.py:706: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
2022-03-10 10:36:26.567860: I tensorflow/core/grappler/devices.cc:69] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
2022-03-10 10:36:26.568099: I tensorflow/core/grappler/clusters/single_machine.cc:357] Starting new session
2022-03-10 10:36:27.789245: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1144] Optimization results for grappler item: graph_to_optimize
  constant_folding: Graph size after: 560 nodes (-640), 1215 edges (-640), time = 822.256ms.
  function_optimizer: function_optimizer did nothing. time = 14.505ms.
  constant_folding: Graph size after: 560 nodes (0), 1215 edges (0), time = 142.14ms.
  function_optimizer: function_optimizer did nothing. time = 8.71ms.

2022-03-10 10:36:28,472 - INFO - Using tensorflow=2.5.0, onnx=1.10.2, tf2onnx=1.9.3/1190aa
2022-03-10 10:36:28,472 - INFO - Using opset <onnx, 9>
2022-03-10 10:36:34,200 - INFO - Computed 0 values for constant folding
2022-03-10 10:36:38,980 - INFO - Optimizing ONNX model
2022-03-10 10:36:43,739 - INFO - After optimization: Add -1 (18->17), BatchNormalization -53 (53->0), Const -162 (270->108), GlobalAveragePool +1 (0->1), Identity -57 (57->0), ReduceMean -1 (1->0), Squeeze +1 (0->1), Transpose -213 (214->1)
2022-03-10 10:36:43,981 - INFO -
2022-03-10 10:36:43,982 - INFO - Successfully converted TensorFlow model resnet_savedmodel to ONNX
2022-03-10 10:36:43,982 - INFO - Model inputs: ['input_1']
2022-03-10 10:36:43,982 - INFO - Model outputs: ['predictions']
2022-03-10 10:36:43,982 - INFO - ONNX model is saved at resnet_savedmodel.onnx
------------------------------------------------ Captured stderr call -------------------------------------------------
2022-03-10 10:35:43.305643: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2022-03-10 10:35:43.305747: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2022-03-10 10:35:49.495800: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2022-03-10 10:35:49.495848: W tensorflow/stream_executor/cuda/cuda_driver.cc:326] failed call to cuInit: UNKNOWN ERROR (303)
2022-03-10 10:35:49.498031: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: ESLT0114
2022-03-10 10:35:49.498358: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: ESLT0114
2022-03-10 10:35:49.498575: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-10 10:35:52.147754: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:176] None of the MLIR Optimization Passes are enabled (registered 2)
2022-03-10 10:36:01.295019: W tensorflow/python/util/util.cc:348] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "C:\Users\ChristiaanMeijer\anaconda3\envs\dianna-x\lib\logging\__init__.py", line 2142, in shutdown
    h.close()
  File "C:\Users\ChristiaanMeijer\AppData\Roaming\Python\Python39\site-packages\absl\logging\__init__.py", line 951, in close
    self.stream.close()
  File "C:\Users\ChristiaanMeijer\anaconda3\envs\dianna-x\lib\site-packages\ipykernel\iostream.py", line 428, in close
    if self._exc:
AttributeError: 'OutStream' object has no attribute '_exc'
```

On the server, converting the model and writing an onnx file fails and I get the following [output](https://github.com/dianna-ai/dianna/runs/5492909059?check_suite_focus=true):
```python
Exception                                 Traceback (most recent call last)
Input In [6], in <cell line: 2>()
      1 s = '\n'.join(x)
----> 2 raise Exception(s)

Exception: 2022-03-10 08:54:22.995977: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2022-03-10 08:54:22.996064: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
C:\hostedtoolcache\windows\Python\3.9.10\x64\lib\runpy.py:127: RuntimeWarning: 'tf2onnx.convert' found in sys.modules after import of package 'tf2onnx', but prior to execution of 'tf2onnx.convert'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
2022-03-10 08:54:25.808464: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2022-03-10 08:54:25.808497: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2022-03-10 08:54:25.811401: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: fv-az89-686
2022-03-10 08:54:25.811663: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: fv-az89-686
2022-03-10 08:54:25.812088: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-10 08:54:31,834 - INFO - Signatures found in model: [serving_default].
2022-03-10 08:54:31,836 - INFO - Output names: ['predictions']
2022-03-10 08:54:31.860454: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
2022-03-10 08:54:31.860827: I tensorflow/core/grappler/clusters/single_machine.cc:358] Starting new session
2022-03-10 08:54:31.948164: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1164] Optimization results for grappler item: graph_to_optimize
  function_optimizer: Graph size after: 1202 nodes (878), 1857 edges (1533), time = 40.887ms.
  function_optimizer: function_optimizer did nothing. time = 0.001ms.

WARNING:tensorflow:From C:\hostedtoolcache\windows\Python\3.9.10\x64\lib\site-packages\tf2onnx\tf_loader.py:706: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
2022-03-10 08:54:36,179 - WARNING - From C:\hostedtoolcache\windows\Python\3.9.10\x64\lib\site-packages\tf2onnx\tf_loader.py:706: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
2022-03-10 08:54:36.367244: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
2022-03-10 08:54:36.367504: I tensorflow/core/grappler/clusters/single_machine.cc:358] Starting new session
2022-03-10 08:54:38.287928: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1164] Optimization results for grappler item: graph_to_optimize
  constant_folding: Graph size after: 562 nodes (-640), 1217 edges (-640), time = 1246.3ms.
  function_optimizer: function_optimizer did nothing. time = 0.003ms.
  constant_folding: Graph size after: 562 nodes (0), 1217 edges (0), time = 208.684ms.
  function_optimizer: function_optimizer did nothing. time = 0.002ms.

2022-03-10 08:54:39,200 - INFO - Using tensorflow=2.8.0, onnx=1.11.0, tf2onnx=1.9.3/1190aa
2022-03-10 08:54:39,200 - INFO - Using opset <onnx, 9>
2022-03-10 08:54:46,175 - INFO - Computed 0 values for constant folding
2022-03-10 08:54:51,141 - INFO - Optimizing ONNX model
---------------------------- Captured stderr call -----------------------------
2022-03-10 08:53:58.999051: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2022-03-10 08:53:58.999174: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2022-03-10 08:54:01.814347: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2022-03-10 08:54:01.814383: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2022-03-10 08:54:01.816258: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: fv-az89-686
2022-03-10 08:54:01.816370: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: fv-az89-686
2022-03-10 08:54:01.816673: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-10 08:54:12.091435: W tensorflow/python/util/util.cc:368] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
Error in atexit._run_exitfuncs:

Traceback (most recent call last):

  File "C:\hostedtoolcache\windows\Python\3.9.10\x64\lib\logging\__init__.py", line 2142, in shutdown
```

The same output (minus the first few lines) in a diff tool:
![image](https://user-images.githubusercontent.com/6087314/157636478-9b05e60f-79e7-47a8-9bbd-8a7ac8cd7e57.png)
I don't see a clear next step in solving this so I'll leave it as it is for now.

**Comment by egpbos** [comment]
Was fixed in ed4316135c44f3af3ada70272d064ac00bd766cf.
